### ISSUE TITLE
Add option to view all keyboard shortucts available in the app

### ISSUE THREAD

**Issue by colinkiama** [elementary/code#874]
## Prerequisites
- [X] I have searched open and closed issues for duplicates.

## Feature
**Is your feature request related to a problem? Please describe.**

I don't know if there's a faster way to do something with the keyboard instead of the mouse because I don't know the keyboard shortcuts available in the app.

**Describe the solution you'd like**

An additional entry under the app's settings called "Keyboard Shortcuts". Selecting it should open a window that displays all the keyboard shortcuts available in the app. We should be able to continue working in code while the keyboard shortcuts window is still open.

**Existing work**

When you press the super key, the elementaryOS shows a list of keyboard shortcuts available. Shortcuts are grouped in sections 
that describe the context of their usage.

**Additional context**
https://elementaryos.stackexchange.com/questions/17256/what-keyboard-shortcuts-are-available-in-code

**Comment by jeremypw** [comment]
I am pretty sure this has been discussed in a more general sense - there should ideally be an easy way all apps can do this with a consistent UX. For example see https://github.com/elementary/shortcut-overlay/issues/53.  Unfortunately its not simple to implement.

**Comment by jeremypw** [comment]
One thing that would be easy to do would be to add a page to the wiki showing the shortcuts. This could be linked to the `F1` key which currently does nothing.

**Comment by colinkiama** [comment]
Yeah, a wiki page would be a great compromise! :+1:

**Comment by jeremypw** [comment]
I have put up a simple page on the wiki listing the shortcuts I found in the code at https://github.com/elementary/code/wiki/shortcuts
If I have got anything wrong or if something changes please let me know.

**Comment by jeremypw** [comment]
Unfortunately the design team were not OK with having a link to a remote website in the app.

**Comment by colinkiama** [comment]
Hmm, what about displaying the shortcuts in a Shortcuts Window?: https://docs.gtk.org/gtk3/class.ShortcutsWindow.html

**Comment by jeremypw** [comment]
@colinkiama Yes, a local solution is probably the only way to fix this in the short term - I was hoping a system-level solution would materialise that all elementary apps could utilise to give a consistent UI but maybe its time to just go ahead with a Gtk.ShortcutsWindow. @danirabbit What do you think?

**Comment by zeebok** [comment]
> I was hoping a system-level solution would materialise that all elementary apps could utilise to give a consistent UI

You mean something in Gala where you could, for example, press Super-H in any app and it lists all of the registered actions and the key board shortcuts? That would be pretty cool.

**Comment by jeremypw** [comment]
@zeebok Yes, something like that but I imagine it would still require each app to actively register the actions/shortcuts with the window manager or maybe implement a Dbus interface. I don't see how it can work transparently. Notwithstanding that. it would ensure a consistent UI.
